src/drivers/mount/mount.cpp

```diff
@@ -0,0 +1,467 @@
+						position_setpoint_triplet->next.lat,
+						position_setpoint_triplet->next.lon,
+						position_setpoint_triplet->next.alt
+					);
+
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_WPINDEX) {
+					//TODO how to do this?
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_LOCATION) {
+					mount_mavlink_point_location(
+						vehicle_global_position->lat,
+						vehicle_global_position->lon,
+						vehicle_global_position->alt,
+						vehicle_roi->lat,
+						vehicle_roi->lon,
+						vehicle_roi->alt
+					);
+
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_TARGET) {
+					//TODO is this even suported?
+				}
+			}
+		}
+
+		mount_mavlink_deinit();
+
+	} else if (mount_state == RC) {
+		if (!mount_rc_init()) {
+			err(1, "could not initiate mount_rc");
+		}
+
+		warnx("running mount driver in rc mode");
+
+		while (!thread_should_exit) {
+			mount_update_topics();
+
+			if (vehicle_roi_updated) {
+				vehicle_roi_updated = false;
+				mount_rc_configure(vehicle_roi->mode, (params.mnt_man_control == 1));
+
+				if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_NONE) {
+					if (params.mnt_man_control && manual_control_setpoint_updated) {
+						//TODO use mount_rc_point_manual to control gimbal
+						//with specified aux channels via the parameters
+					}
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_WPNEXT) {
+					mount_rc_set_location(
+						vehicle_global_position->lat,
+						vehicle_global_position->lon,
+						vehicle_global_position->alt,
+						position_setpoint_triplet->next.lat,
+						position_setpoint_triplet->next.lon,
+						position_setpoint_triplet->next.alt
+					);
+
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_WPINDEX) {
+					//TODO how to do this?
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_LOCATION) {
+					mount_rc_set_location(
+						vehicle_global_position->lat,
+						vehicle_global_position->lon,
+						vehicle_global_position->alt,
+						vehicle_roi->lat,
+						vehicle_roi->lon,
+						vehicle_roi->alt
+					);
+
+				} else if (vehicle_roi->mode == vehicle_roi_s::VEHICLE_ROI_TARGET) {
+					//TODO is this even suported?
+				}
+			}
+
+			mount_rc_point();
+		}
+
+		mount_rc_deinit();
+
+	} else if (mount_state == ONBOARD) {
+		if (!mount_onboard_init()) {
+			err(1, "could not initiate mount_onboard");
+		}
+
+		warnx("running mount driver in onboard mode");
+
+		while (!thread_should_exit) {
+			mount_update_topics();
+			mount_onboard_update_topics();
+
+			if (vehicle_mount_updated) {
+				if (vehicle_mount->config) {
+					mount_onboard_configure(vehicle_mount->mode,
+								vehicle_mount->stab_roll, vehicle_mount->stab_pitch, vehicle_mount->stab_yaw);
+
+				} else {
+					switch (vehicle_mount->mode) {
+					case vehicle_mount_s::VEHICLE_MOUNT_MODE_RETRACT:
+					case vehicle_mount_s::VEHICLE_MOUNT_MODE_NEUTRAL:
+						break;
+
+					case vehicle_mount_s::VEHICLE_MOUNT_MODE_MAVLINK_TARGETING:
```

Comment:
MAVLink and RC targeting: in the current app we have the possibility to override the mount-mode with an AUX switch. The reason for implementing it like this was to be able to set the gimbal and retracts to locked/netural position in-case what ever controls the gimbal via MAVLink fails and you need to land manually.
So you would need another param to specify a mount-mode switch and add a similar override handling (depending on the vehicle control mode). At the same time this can be used in full manual flight to control the retracts.